tests/fixtures/components/dataset_activity.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (C) 2022 Indoc Research
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import random
+from datetime import datetime
+from typing import Any
+
+import pytest
+
+from search.components import DatasetActivity
+from search.components import ModelList
+from search.components.dataset_activity.crud import DatasetActivityCRUD
+from search.components.dataset_activity.schemas import DatasetActivitySchema
+from tests.fixtures.components._base_factory import BaseFactory
+
+
+class DatasetActivityFactory(BaseFactory):
+    """Create dataset activity related entries for testing purposes."""
+
+    def generate(  # noqa: C901
+        self,
+        activity_type: str = ...,
+        activity_time: datetime = ...,
+        container_code: str = ...,
+        version: str = ...,
+        target_name: str = ...,
+        user: str = ...,
+        changes: list[dict[str, Any]] = ...,
+    ) -> DatasetActivitySchema:
+
+        if activity_type is ...:
+            activity_type = random.choice(['create', 'update', 'schema_update', 'publish'])
+
+        if activity_time is ...:
+            activity_time = self.fake.past_datetime()
+
+        if container_code is ...:
+            container_code = self.fake.unique.word().lower()
+
+        if version is ...:
+            version = f'{self.fake.pyint(0, 4)}.{self.fake.pyint(0, 20)}'
+
+        if target_name is ...:
+            target_name = '.'.join(self.fake.words(3)).lower()
+
+        if user is ...:
+            user = self.fake.word().lower()
```

Comment:
I suggest to add here more uniqueness, because sometimes same word can be generated two times and some tests that expect exact value in filtering will fail.
```suggestion
            user = f'{self.fake.pystr()}-{self.fake.first_name()}'.lower()
```